contracts/core/funds-distribution-token/v1/ExtendedFundsTokenFDT.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.6.11;
+
+import { IERC20, SafeERC20 } from "../../../../lib/openzeppelin-contracts/contracts/token/ERC20/SafeERC20.sol";
+
+import { ExtendedFDT, SafeMath, SignedSafeMath } from "../../funds-distribution-token/v1/ExtendedFDT.sol";
+
+import { IBasicFDT }              from "./interfaces/IBasicFDT.sol";
+import { IExtendedFundsTokenFDT } from "./interfaces/IExtendedFundsTokenFDT.sol";
+
+/// @title ExtendedFundsTokenFDT implements the FDT functionality for accounting for losses with a separate Funds Token.
+abstract contract ExtendedFundsTokenFDT is IExtendedFundsTokenFDT, ExtendedFDT {
```

Comment:
if we already adding `FundsToken` in the name then we don't need FDT as the suffix

Reply:
The issue is that `FundsToken` implies that the funds gains or lost are in a different type of token: the `fundsToken`, whereas FDT implies that funds are generally distributed based on token ownership, but says nothing about whether that distribution is of the same token, of of a `fundsToken`. Honestly, all the naming (contracts, functions, and variables) are pretty terrible across out FDTs. I am itching to standardize and refactor it all.

Reply:
We should have something like:
- `GainsDistributionToken`
- `LossesDistributionToken`
- `GainsAndLossesDistributionToken`
- `SecondaryGainsDistributionToken`
- `SecondaryLossesDistributionToken`
- `SecondaryGainsAndLossesDistributionToken`